library-udf/src/test/java/org/apache/iotdb/library/string/StringTests.java

```diff
@@ -0,0 +1,208 @@
+        TSDataType.TEXT,
+        TSEncoding.PLAIN,
+        CompressionType.UNCOMPRESSED,
+        null);
+  }
+
+  private static void generateData() {
+    try (Connection connection =
+            DriverManager.getConnection(
+                Config.IOTDB_URL_PREFIX + "127.0.0.1:6667/", "root", "root");
+        Statement statement = connection.createStatement()) {
+      statement.execute(
+          String.format(
+              "insert into root.vehicle.d1(timestamp,s1) values(%d,%s)",
+              100, "[192.168.0.1] [SUCCESS]"));
+      statement.execute(
+          String.format(
+              "insert into root.vehicle.d1(timestamp,s1) values(%d,%s)",
+              200, "[192.168.0.24] [SUCCESS]"));
+      statement.execute(
+          String.format(
+              "insert into root.vehicle.d1(timestamp,s1) values(%d,%s)",
+              300, "[192.168.0.2] [FAIL]"));
+      statement.execute(
+          String.format(
+              "insert into root.vehicle.d1(timestamp,s1) values(%d,%s)",
+              400, "[192.168.0.5] [SUCCESS]"));
+      statement.execute(
+          String.format(
+              "insert into root.vehicle.d1(timestamp,s1) values(%d,%s)",
+              500, "[192.168.0.124] [SUCCESS]"));
+      statement.execute(
+          String.format(
+              "insert into root.vehicle.d2(timestamp,s1) values(%d,%s)", 100, "A,B,A+,B-"));
+      statement.execute(
+          String.format(
+              "insert into root.vehicle.d2(timestamp,s1) values(%d,%s)", 200, "A,A+,A,B+"));
+      statement.execute(
+          String.format("insert into root.vehicle.d2(timestamp,s1) values(%d,%s)", 300, "B+,B,B"));
+      statement.execute(
+          String.format(
+              "insert into root.vehicle.d2(timestamp,s1) values(%d,%s)", 400, "A+,A,A+,A"));
+      statement.execute(
+          String.format(
+              "insert into root.vehicle.d2(timestamp,s1) values(%d,%s)", 500, "A,B-,B,B"));
+    } catch (SQLException throwable) {
+      fail(throwable.getMessage());
+    }
+  }
+
+  private static void registerUDF() {
+    try (Connection connection = EnvFactory.getEnv().getConnection();
+        Statement statement = connection.createStatement()) {
+      statement.execute(
+          "create function regexmatch as 'org.apache.iotdb.library.string.UDTFRegexMatch'");
+      statement.execute(
+          "create function regexreplace as 'org.apache.iotdb.library.string.UDTFRegexReplace'");
+      statement.execute(
+          "create function regexsplit as 'org.apache.iotdb.library.string.UDTFRegexSplit'");
+      statement.execute(
+          "create function strreplace as 'org.apache.iotdb.library.string.UDTFStrReplace'");
+    } catch (SQLException throwable) {
+      fail(throwable.getMessage());
+    }
+  }
+
+  @AfterClass
+  public static void tearDown() throws Exception {
+    EnvFactory.getEnv().cleanAfterClass();
+    ConfigFactory.getConfig()
+        .setUdfCollectorMemoryBudgetInMB(100)
```

Comment:
It's better to use variables storing the previous config values in `setUp()` and reset them in `tearDown()`. Otherwise this change may influent other following test cases.

Reply:
Done.